### PR TITLE
Improve scoping of '$current_test' variable

### PR DESCRIPTION
**Pull request by okurz**
This should help with introducing a stricter programming style check.

### CODE REVIEW

File: autotest.pm [hunk 1]
```diff
@@ -295,33 +295,6 @@ sub start_process {
     return ($process, $child);
 }
 
-
-# TODO: define use case and reintegrate
-sub prestart_hook {
```
Comment (Martchus) [inline autotest.pm:300]: Aren't these functions strictly a part of the API? It looks like they are not used by our test distribution and likely also not elsewhere, though.

Comment (okurz) [reply autotest.pm:300]: I think we only consider testapi.pm and *maybe* distribution.pm as part of the official, public API that should use deprecation before removal

Comment (Martchus) [reply autotest.pm:300]: right

**Review by Martchus — APPROVED** [review]
Tidy doesn't like it but otherwise it should be fine.

**Pull request merged**